File: packages/android-sdk/sdk/src/main/kotlin/com/opencom/sdk/OpencomConfig.kt
package com.opencom.sdk

/**
 * Configuration for the Opencom SDK.
 *
 * @property workspaceId Your Opencom workspace ID
 * @property convexUrl The Convex backend URL
 * @property theme Optional custom theme for UI components
 * @property debug Enable debug logging (default: false)
 */
data class OpencomConfig(
    val workspaceId: String,
    val convexUrl: String,
    val theme: OpencomTheme = OpencomTheme(),
    val debug: Boolean = false
)

File: packages/ios-sdk/Sources/OpencomSDK/Core/OpencomConfig.swift
import Foundation

/// Configuration for the Opencom SDK.
public struct OpencomConfig {
    /// The workspace ID for your Opencom account
    public let workspaceId: String

    /// The Convex backend URL
    public let convexUrl: String

    /// Theme customization for UI components
    public let theme: OpencomTheme

    /// Enable debug logging
    public let debug: Bool

    public init(
        workspaceId: String,
        convexUrl: String,
        theme: OpencomTheme = OpencomTheme(),
        debug: Bool = false
    ) {
        self.workspaceId = workspaceId
        self.convexUrl = convexUrl
        self.theme = theme
        self.debug = debug
    }
}
